### PR TITLE
Bug 1552013	- Update README in browser/components/newtab to reflect mc-based workflow

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

File: README-mc.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Firefox Home (New Tab)
```
Comment (k88hudson) [inline README-mc.md]: Not sure if there's a better name for this file than `README-mc.md`

File: README-mc.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+This directory contains some source files (jsx, sass) that need to be built outside of the `./mach` build system.
+We are working on migrating this build system to work with `mach`. In the meantime, please follow the following
+steps if you need to make changes to code in this directory:
```
Comment (Mardak) [inline README-mc.md]: Maybe a note that the commands are run from m-c root directory where one would normally run `./mach` And perhaps a note of going directly into `browser/components/newtab` to skip the `--prefix …`

**Comment by Mardak** [comment]
Oh, and if we do the SPHINX moz.build stuff, theoretically the doc should just appear at https://firefox-source-docs.mozilla.org/

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+You will need the following:
+
+- Node.js 8+ (On Mac, the best way to install Node.js is to use the [install link on the Node.js homepage](https://nodejs.org/en/))
+- npm (packaged with Node.js)
+
+To install dependencies, run the following from the root of the mozilla-central repository
+(or cd into browser/components/newtab to omit the `--prefix` in any of these commands):
+
+```
+npm install --prefix browser/components/newtab
```
Comment (Mardak) [inline docs/index.rst]: You can use `.. code-block:: shell` (followed by a new line and indented code)

E.g., https://hg.mozilla.org/try/diff/a10de97f521abc690c33f16186965d6d65535fe5/browser/components/newtab/docs/index.rst#l1.34

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+```````````````````````````````````
+
+To build assets and run Firefox, run the following from the root of the mozilla-central repository:
+
+```
+npm run bundle --prefix browser/components/newtab && ./mach run
+```
+
+Running tests
+`````````````````
```
Comment (Mardak) [inline docs/index.rst]: nit: extra `s

File: .mcignore [hunk 5]
```diff
@@ -14,6 +14,9 @@ npm-debug.log
 /logs/
 /node_modules/
 
+# ignore README since it's GitHub specific
+/README.md
```
Comment (Mardak) [inline .mcignore]: We don't really need to remove it but I guess it's fine if we do. But if we do keep it in mozilla-central, we could note that the doc is GitHub focused and to see https://firefox-source-docs.mozilla.org/browser/components/newtab/docs/ (assuming we name the url path "docs" with `SPHINX_TREES['docs'] = 'docs'` where the left "docs" is the url path and the right "docs" is our directory structure)

**Pull request merged**